pkg/options/options.go

```diff
@@ -0,0 +1,60 @@
+		// shared options
+		Overwrite bool `help:"Whether to overwrite any existing symbol file with a matching ID"`
+		Timeout   int  `help:"Number of seconds to wait before failing an upload request" default:"300"`
+		Retries   int  `help:"Number of retry attempts before failing an upload request" default:"0"`
+
+		// required options
+		AndroidAab         upload.AndroidAabMapping      `cmd:"" help:"Process and upload application bundle files for Android"`
+		All                upload.DiscoverAndUploadAny   `cmd:"" help:"Upload any symbol/mapping files"`
+		AndroidNdk         upload.AndroidNdkMapping      `cmd:"" help:"Process and upload Proguard mapping files for Android"`
+		AndroidProguard    upload.AndroidProguardMapping `cmd:"" help:"Process and upload NDK symbol files for Android"`
+		DartSymbol         upload.DartSymbolOptions      `cmd:"" help:"Process and upload symbol files for Flutter" name:"dart"`
+		ReactNativeAndroid upload.ReactNativeAndroid     `cmd:"" help:"Upload source maps for React Native Android"`
+		UnityAndroid       upload.UnityAndroid           `cmd:"" help:"Upload Android mappings and NDK symbol files from Unity projects"`
+	} `cmd:"" help:"Upload symbol/mapping files"`
+	CreateBuild          CreateBuild          `cmd:"" help:"Provide extra information whenever you build, release, or deploy your application"`
+	CreateAndroidBuildId CreateAndroidBuildId `cmd:"" help:"Generate a reproducible Build ID from .dex files"`
+}
+
+type CreateAndroidBuildId struct {
+	Path utils.Paths `arg:"" name:"path" help:"Path to the project directory" type:"path"`
+}
+
+type CreateBuild struct {
+	BuilderName       string            `help:"The name of the entity that triggered the build. Could be a user, system etc."`
+	Metadata          map[string]string `help:"Additional build information"`
+	ReleaseStage      string            `help:"The release stage (eg, production, staging) that is being released (if applicable)."`
+	Provider          string            `help:"The name of the source control provider that contains the source code for the build."`
+	Repository        string            `help:"The URL of the repository containing the source code being deployed."`
+	Revision          string            `help:"The source control SHA-1 hash for the code that has been built (short or long hash)"`
+	Path              utils.Paths       `arg:"" name:"path" help:"Path to the project directory" type:"path" default:"."`
+	VersionName       string            `help:"The version of the application." xor:"app-version,version-name"`
+	VersionCode       string            `help:"The version code for the application (Android only)." xor:"app-version-code,version-code"`
+	BundleVersion     string            `help:"The bundle version for the application (iOS only)." xor:"app-bundle-version,bundle-version"`
+	AutoAssignRelease bool              `help:"TBC"`
+	AppManifest       utils.Path        `help:"TBC"`
+	AndroidAab        utils.Path        `help:"TBC"`
```

Comment:
Can the Android-specific options be nested into a `build.Android` struct (or similar) to logically separate them? Once we have iOS options and potential future ones, I think keeping them separate will be useful.

Reply:
Should be possible.